Plugins/NativeEngine/Source/VertexArray.cpp

```diff
@@ -85,7 +95,50 @@ namespace Babylon
 
     void VertexArray::SetVertexBuffers(bgfx::Encoder* encoder, uint32_t startVertex, uint32_t numVertices)
     {
-        for (auto& pair : m_vertexBufferRecords)
+        const bgfx::Caps* caps = bgfx::getCaps();
+
+        // Check if instancing is supported.
+        const bool instancingSupported = 0 != (BGFX_CAPS_INSTANCING & caps->supported);
+        if (!m_vertexBufferInstanceRecords.empty() && instancingSupported)
+        {
+            bgfx::InstanceDataBuffer instanceDataBuffer{};
+
+            // instance stride
+            uint16_t instanceStride{};
+            uint32_t instanceCount{};
+            for (auto& pair : m_vertexBufferInstanceRecords)
+            {
+                const auto vertexBuffer{pair.second.Buffer};
+                instanceCount = static_cast<uint32_t>(vertexBuffer->GetBytes().size()) / pair.second.Stride;
+                instanceStride += static_cast<uint16_t>(pair.second.ElementSize);
+            }
+
+            // create instance datas. Instance Data Buffer is transient.
+            bgfx::allocInstanceDataBuffer(&instanceDataBuffer, instanceCount, instanceStride);
+
+            // copy instance datas
+            uint8_t* data{instanceDataBuffer.data};
+            uint32_t offset{};
+
+            // reverse because of bgfx also reverting : https://github.com/bkaradzic/bgfx/blob/4581f14cd481bad1e0d6292f0dd0a6e298c2ee18/src/renderer_d3d11.cpp#L2701
+#if D3D11 || D3D12 
+            for (auto iter = m_vertexBufferInstanceRecords.rbegin(); iter != m_vertexBufferInstanceRecords.rend(); ++iter)
+#else
+            for (auto iter = m_vertexBufferInstanceRecords.cbegin(); iter != m_vertexBufferInstanceRecords.cend(); ++iter)
+#endif
```

Comment:
would it be okay to always reverse?

Reply:
I tried to find a solution for this without success. Because of attribute types ordering and bgfx, I didn't find a solution that fixes it.

Reply:
Is the answer to my question no then?

Reply:
yes, it's no.